### ISSUE TITLE
[Question] Is ngrok / pyngrok still not valid in google colab enviroment?

### ISSUE THREAD

**Issue by timtensor** [ngrok/ngrok-python#100]
I tried the colab notebook to use google colab together with my vscode environment.  Is it still banned because i failed to connect it to the notebook machine.

**Comment by salilsub** [comment]
Hi @timtensor . Colab is not blocked on our end, could you share some additional detail about what you are trying to do and what kinds of errors/issues you are running into?

**Comment by timtensor** [comment]
Actually i was trying to run vscode via pyngrok to run my local code in remote server. Everytime pyngrok runs, colab mentions that is not permitted and gets disconnected.

**Comment by timtensor** [comment]
Is there a way to run vscode remotely on ngrok ?

**Comment by salilsub** [comment]
Hi @timtensor apologies for the delayed response. So pyngrok is not an officially supported ngrok library, for that we recommend [ngrok-python](https://github.com/ngrok/ngrok-python). However, pyngrok is using our agent under the hood, so there are a few things that could be happening:
* We require that you have a valid authtoken to connect to ngrok. In the past you could connect without authenticating so this could be a reason.
* We have an [agent version support policy](https://ngrok.com/docs/agent/version-support-policy/) for free accounts. Pyngrok installs a very old version of ngrok by default. You might need to upgrade to a more recent version to make it work.

With respect to using VSCode and ngrok, we have a [guide](https://ngrok.com/docs/using-ngrok-with/vsCode)

**Comment by russorat** [comment]
@timtensor thanks for the issue. I'm going to close this as stale. if you are still having issues with running in colab, please open a new issue with updated details.
